lib/tlscontext.c

```diff
@@ -334,7 +335,25 @@ tls_context_setup_session(TLSContext *self)
         }
 
       SSL_CTX_set_verify(self->ssl_ctx, verify_mode, tls_session_verify_callback);
-      SSL_CTX_set_options(self->ssl_ctx, SSL_OP_NO_SSLv2);
+      if (self->ssl_options != TSO_NONE)
+        {
+          ssl_options=0;
+          if(self->ssl_options & TSO_NOSSLv2)
+            ssl_options |= SSL_OP_NO_SSLv2;
+          if(self->ssl_options & TSO_NOSSLv3)
+            ssl_options |= SSL_OP_NO_SSLv3;
+          if(self->ssl_options & TSO_NOTLSv1)
+            ssl_options |= SSL_OP_NO_TLSv1;
+#ifdef SSL_OP_NO_TLSv1_2
+          if(self->ssl_options & TSO_NOTLSv11)
+            ssl_options |= SSL_OP_NO_TLSv1_1;
+          if(self->ssl_options & TSO_NOTLSv12)
+            ssl_options |= SSL_OP_NO_TLSv1_2;
+#endif
+          SSL_CTX_set_options(self->ssl_ctx, ssl_options);
+        }
+      else
+	msg_debug("empty ssl options",NULL);
```

Comment:
As far as I can see we *transform* our `ssl_options` into `openssl_options`. This transformation could be carried out by declaring  a `compile_table` where the entries are `{ssl_option, openssl_option}` pairs. After that a simple iteration through the elements can do the work.

I also support the idea of @lbudai of declaring `TLSSSLOption` and `TLSOpenSSLOption`.

Sketch:
```C

typedef gint TLSSSLOption;
typedef gint TLSOpenSSLOption;

typdef struct 
{
   TLSSSLOption ssl_option;
   TLSOpenSSLOption openssl_option;
} TLSSSLOptionPair;

TLSSSLOptionPair TLSSSLOptionTransformTable[]= 
{
     { TSONOSSLv2, SSL_OP_N_SSLv2 },
     ...
     ...
     ...
};

........

`ssl_transform_options function`:

for(int i = 0; i < sizeof(TLSSSLOptionTransformTable); ++i)
{
    if ( self->ssl_options & TLSSSLOptionTransformTable[i].ssl_option )
        ssl_options |= TLSSLOptionTransformTable[i].openssl_option;
}

```